Core/src/Geometry/GridPortalLinkMerging.cpp

```diff
@@ -0,0 +1,569 @@
+          (axisA.getMax() - axisA.getMin() + axisB.getMax() - axisB.getMin()) /
+          2.0;
+
+      min = -halfWidth;
+      max = halfWidth;
+    }
+
+    return makeGrid(mergedSurface, direction, logger,
+                    std::tuple{min, max, axisA.getNBins() + axisB.getNBins()},
+                    otherAxis, prepend);
+
+  } else {
+    ACTS_VERBOSE("==> binWidths differ: " << binsWidthA << " vs " << binsWidthB
+                                          << " ~> variable merge");
+
+    std::unique_ptr<GridPortalLink> mergedPortalLink =
+        mergeVariable(mergedSurface, axisA, axisB, tolerance, direction, logger,
+                      otherAxis, prepend);
+    return mergedPortalLink;
+  }
+}
+
+std::unique_ptr<GridPortalLink> colinearMerge(
+    const std::shared_ptr<RegularSurface>& mergedSurface, const IAxis& axisA,
+    const IAxis& axisB, ActsScalar tolerance, BinningValue direction,
+    const Logger& logger, const IAxis* otherAxis, bool prepend) {
+  AxisType aType = axisA.getType();
+  AxisType bType = axisB.getType();
+  if (axisA.getBoundaryType() != axisB.getBoundaryType()) {
+    ACTS_WARNING("AxisBoundaryTypes are different");
+    return nullptr;
+  }
+
+  if (axisA.getBoundaryType() != AxisBoundaryType::Bound) {
+    ACTS_WARNING("AxisBoundaryType is not Bound, cannot do colinear merge");
+    return nullptr;
+  }
+
+  // convenience only
+  auto mergeVariableLocal = [&] {
+    return mergeVariable(mergedSurface, axisA, axisB, tolerance, direction,
+                         logger, otherAxis, prepend);
+  };
+
+  if (aType == AxisType::Equidistant && bType == AxisType::Equidistant) {
+    auto mergedPortalLink =
+        mergeEquidistant(mergedSurface, axisA, axisB, tolerance, direction,
+                         logger, otherAxis, prepend);
+    return mergedPortalLink;
+  } else if (aType == AxisType::Variable && bType == AxisType::Variable) {
+    ACTS_VERBOSE("===> variable merge");
+    auto mergedPortalLink = mergeVariableLocal();
+    return mergedPortalLink;
+  } else if (aType == AxisType::Equidistant && bType == AxisType::Variable) {
+    ACTS_WARNING("===> mixed merged");
+    auto mergedPortalLink = mergeVariableLocal();
+    return mergedPortalLink;
+  } else {
+    ACTS_WARNING("===> mixed merged");
+    auto mergedPortalLink = mergeVariableLocal();
+    return mergedPortalLink;
+  }
+}
+
+std::unique_ptr<PortalLinkBase> mergeGridPortals(
+    const GridPortalLink* a, const GridPortalLink* b,
+    const RegularSurface* surfaceA, const RegularSurface* surfaceB,
+    BinningValue loc0, BinningValue loc1, BinningValue direction,
+    const Logger& logger) {
+  assert(surfaceA != nullptr);
+  assert(surfaceB != nullptr);
+  assert(surfaceA->type() == surfaceB->type());
+  assert(a->dim() == 2 || a->dim() == 1);
+  assert(a->dim() == b->dim());
+
+  ACTS_VERBOSE(" - a: " << a->surface().bounds());
+  ACTS_VERBOSE(" - b: " << b->surface().bounds());
+
+  constexpr auto tolerance = s_onSurfaceTolerance;
```

Comment:
I use this tolerance to check for equivalent bin widths and edges further down the call chain.

I don't believe this needs to be configurable, the reason I set this to `s_onSurfaceTolerance` is only because I didn't want to hard-code it to an arbitrary number here, but still have it defined in a single place.

Reply:
makes sense 👍 should we add a small comment that this is intended to not be configurable?